### PR TITLE
Test: stabilize study options click test

### PR DESCRIPTION
**Pull request by mikehardy**
- different selectors for elements (so it doesn't flake on Android 5)
- keyboard dismiss when entering note editor (so it doesn't flake Android 8)
- note type selection (so it doesn't flake in general)

Related #9073

## How Has This Been Tested?

Running `./gradlew jacocoAndroidTestReport` over and over while API21-API31 phone + API30 Pixel C tablet emulators were up

## Learning (optional, can help others)

instrumented tests are subtle, but the error messages aren't too bad to follow towards fixes

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by mikehardy** [comment]
In my mind this is still targeted for release-2.15 so it's a very high priority for me + it stabilizes emulator test on main branch.
Or at least I hope it does. It should?

**Comment by mikehardy** [comment]
RootView did not appear in time on this, so it's still not right in CI. Better locally but I dunno.

**Comment by mikehardy** [comment]
Yep. Still work to do

```

com.ichi2.anki.DeckPickerTest > checkIfClickOnCountsLayoutOpensStudyOptionsOnMobile[TestingAVD(AVD) - 11] FAILED 
	androidx.test.espresso.base.RootViewPicker$RootViewWithoutFocusException: Waited for the root of the view hierarchy to have window focus and not request layout for 10 seconds. If you specified a non default root matcher, it may be picking a root that never takes focus. Root:
	Root{application-window-token=android.view.ViewRootImpl$W@adf7935, window-token=android.view.ViewRootImpl$W@adf7935, has-window-focus=false, layout-params-type=1, layout-params-string={(0,0)(fillxfill) sim={forwardNavigation} ty=BASE_APPLICATION fmt=RGBA_8888 wanim=0x10302fe

```

At least they actually run locally now, I know there's progress.

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/DeckPickerTest.java [hunk 1]
```diff
@@ -101,10 +101,17 @@ private void createDeckWithCard(String testString) {
...
 
-        // Create a card belonging to the new deck
+        // Create a card belonging to the new deck, using Basic type (guaranteed to exist)
         onView(withId(R.id.fab_main)).perform(click());
         onView(withId(R.id.add_note_action)).perform(click());
-        onView(withIndex(withId(R.id.id_note_editText), 0)).perform(typeText("SampleText" + testString));
+
+        // Close the keyboard, it auto-focuses and obscures enough of the screen
+        // on some devices that espresso complains about global visibility being <90%
+        pressBack();
```
Comment (david-allison-1) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/DeckPickerTest.java:110]: Sorry for the nitpick, could you extract this to a `closeKeyboard` method.

My thoughts:

* Android is moving to a nicer keyboard API
* We may decide to remove "double press back to exit", in which case we'll want a better method here (likely involving detecting the keyboard to be open, and knowing where it's used will save time)

**Comment by mikehardy** [comment]
Unable to reproduce locally when I run the same emulator create / emulator start / test command lines that run in CI, and the environment difference between CI and my local environment is mostly processing power which implies timing etc. My guess is that there is a system ANR screen or similar showing due to the slower CI environment and it confuses espresso.

That implies that a bisect locally will work with this commit though.

There are some well-known strategies for either ANR dismiss or for specifying a different window root via espresso RootMatchers and that will be the next step here to move this forward but it will have to sit for a while unfortunately, this weekend I had a lot of time and this week I will have very little.

**Comment by mikehardy** [comment]
This is the example on how to specify the Root View to match instead of relying on the espresso heuristic - https://developer.android.com/training/testing/espresso/recipes#targeting-non-default-windows

**Comment by mikehardy** [comment]
This test flake is killing me ;-) - I have to manually check the CI runs every time.
Ignoring it in CI now pending an issue I will peel off about stabilizing root view matching, but now it runs successfully locally and won't gum up CI runs.

There was already a method for closing the softkeyboard in Espresso as a static, so just used that for keyboard closing :+1:

**Review by david-allison-1 — APPROVED** [review]
I was considering this for a while, most definitely (as long as there's a linked issue)

**Comment by david-allison-1** [comment]
@mikehardy Awaiting a linked issue (can't find one, https://github.com/ankidroid/Anki-Android/issues/9056 is closed). Ideally linked to the string in the commit/commit description

**Comment by mikehardy** [comment]
done: peeled issue, referenced in ignore strings + commit message

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Pull request merged**